jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -441,9 +462,8 @@ BuildConfiguration toBuildConfiguration(Containerizer containerizer)
         .setContainerConfiguration(containerConfigurationBuilder.build())
         .setLayerConfigurations(layerConfigurations)
         .setAllowInsecureRegistries(containerizer.getAllowInsecureRegistries())
-        .setToolName(containerizer.getToolName());
-
-    containerizer.getExecutorService().ifPresent(buildConfigurationBuilder::setExecutorService);
+        .setToolName(containerizer.getToolName())
+        .setExecutorService(executorService);
```

Comment:
Should we remove the code calling `newCachedThreadPool()` in `BuildConfiguration`, and return an error if the executor is not set? (`BuildConfiguration` is not public-facing.)

```
          if (executorService == null) {
            executorService = Executors.newCachedThreadPool();
          }
```

Reply:
Oops, that had been removed, but I reworked the patch and it was restored.